lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -203,7 +207,57 @@ namespace Js
             return instance;
         }
     };
-#endif
+
+#elif INTL_ICU
```

Comment:
#elif doesn't semantically match #ifdef

Could change both to use `#if defined(...)` and `#elif defined(...)`

I've noticed you tending to depend on the fact that `INTL_ICU` is set to something non-zero as opposed to being defined. My setting it to `1` was defensive against this kind of conflation of defined-ness versus truthy-ness, but `#ifdef / #if defined(...)` is more defensive in general. We know that if this macro is defined it tells us we're using the feature. It's actual value (or lack thereof) should not be important (or could theoretically become important in the future, but unlikely). I'm more concerned that someone will see the `1` and think this is strange style since that `1` has no semantic meaning as a constant and could end up breaking an uncommon config (`INTL_ICU`) -- another reason to get some CI coverage but that's another story.

Reply:
Noted, I can switch this back to elif defined()